presto-main/src/main/java/io/prestosql/event/QueryMonitor.java

```diff
@@ -309,15 +322,34 @@ private QueryContext createQueryContext(SessionRepresentation session, Optional<
 
     private static QueryIOMetadata getQueryIOMetadata(QueryInfo queryInfo)
     {
+        Multimap<QualifiedObjectName, OperatorStats> tableOperatorStats = extractTableScanOperatorStats(queryInfo);
+
         ImmutableList.Builder<QueryInputMetadata> inputs = ImmutableList.builder();
         for (Input input : queryInfo.getInputs()) {
+            // Note: input table may be scanned multiple times
```

Comment:
If two inputs for the same table are added with different columns, then this logic will add the same aggregate value for both the inputs right? Should we add a plan node id to `Input`instead and map the stats here appropriately?

Reply:
You are right.  This will more difficult...  I added fragment id and plan node id to Input so we can find the exact node in the exact fragment.